src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -297,6 +350,46 @@ private IContainer calculateGlobalPKAnalysisFor(Simulation simulation, Compound
          return container;
       }
 
+      private IParameter cmaxRatioCalculator(int individualId, CompoundPK compoundPKFromContext, float? cMax)
+      {
+         var contextValue = compoundPKFromContext.QuantityFor(individualId, CoreConstants.PKAnalysis.C_maxRatio);
+         if (contextValue.HasValue)
```

Comment:
There's some repetition here that can maybe get factored out in these three functions. The idea is that you first check if there's a QuantityPKParameter for the individual already in the context and if so, just use this value.

Otherwise you check if there's a denominator available to calculate the ratio parameter, otherwise it's just null